js/script.js

```diff
@@ -0,0 +1,187 @@
+    random: Math.floor(Math.random() * 3) + 1,
+    score: document.querySelector("#score"),
+    humanScore: 0,
+    computerScore: 0,
+    game: document.querySelector("#new-game"),
+    question: 10,
+    roundNumber: document.querySelector("#r-number"),
+    gameResult: document.querySelector("#g-result"),
+    circleButton: document.querySelectorAll(".circle-btn"),
+    progress: [],
+    eachRound: 0,
+    table: document.querySelector("table"),
+};
+
+//START GAME (registerAddEventListener)
+
+initialize();
+
+function initialize() {
+    registListener();
+    controllButtons(true);
+}
+
+function registListener() {
+    var buttonList = document.querySelectorAll('.player-move');
+
+    for (let i = 0; i < buttonList.length; i++) {
+        const buttonName = buttonList[i].getAttribute('data-move');
+
+        buttonList[i].addEventListener('click', function () {
+            playerMove(buttonName);
+        });
+    }
+}
+
+function controllButtons(disable) {
+    params.circleButton.forEach(function (currentElement) {
+        if (disable) {
+            currentElement.setAttribute('disabled', true);
+            currentElement.classList.remove('circle')
+        } else {
+            currentElement.removeAttribute('disabled');
+            currentElement.classList.add('circle')
+        }
+    });
+}
+
+//NEW GAME
+
+params.game.addEventListener('click', function () {
+    controllButtons(false);
+    resetGame();
+    params.question = window.prompt('How many round would you like to play?');
+    if (isNaN(params.question) || params.question.length < 1) {
+        params.roundNumber.innerHTML = 'Please, write a number';
+    } else {
+        params.winner = Math.floor(params.question * 50 / 100 + 1);
+        params.roundNumber.innerHTML = 'Number of round: ' + params.question + '  -   you will be a winner if you win ' + params.winner + ' rounds';
+    }
+    hideModal();
+});
+
+//RESET GAME
+
+function resetGame() {
+    params.humanScore = 0;
+    params.computerScore = 0;
+    params.score.innerHTML = params.humanScore + ' - ' + params.computerScore;
+    params.output.innerHTML = '';
+    params.gameResult.innerHTML = '';
+    params.eachRound = 0;
+    for (var i = 0; i < params.circleButton.length; i++) {
+        params.circleButton[i].classList.remove('circle');
+    }
+}
+
+function computerMove() {
+    var random = Math.floor(Math.random() * 3);
+    var computerChoices = ['paper', 'rock', 'scissors'];
+    return computerChoices[random];
+};
+
+//PROCESS OF GAME
+
+function playerMove(name) {
+    var computerM = computerMove();
+    params.eachRound++;
+    if ((name === 'rock' && computerM === 'paper') ||
+        (name === 'paper' && computerM === 'scissors') ||
+        (name === 'scissors' && computerM === 'rock')) {
+        params.computerScore++;
+        params.score.innerHTML = params.humanScore + ' - ' + params.computerScore;
+        var computerWon = 'computer won';
```

Comment:
nie ma potrzeby deklarować tej zmiennej skoro używasz jej tylko raz i to kilka linijek niżej